src/main/kotlin/org/malachite/estella/services/TaskService.kt

```diff
@@ -5,42 +5,72 @@ import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import org.malachite.estella.commons.EStellaService
 import org.malachite.estella.commons.UnauthenticatedException
+import org.malachite.estella.commons.decodeBase64
 import org.malachite.estella.commons.models.tasks.Task
 import org.malachite.estella.commons.models.tasks.TaskResult
-import org.malachite.estella.commons.models.tasks.TaskStage
-import org.malachite.estella.process.domain.TaskDto
-import org.malachite.estella.process.domain.TaskTestCaseDto
-import org.malachite.estella.process.domain.toTask
-import org.malachite.estella.process.domain.toTaskDto
+import org.malachite.estella.commons.toBase64String
+import org.malachite.estella.process.domain.*
 import org.malachite.estella.queues.utils.TaskResultRabbitDTO
+import org.malachite.estella.task.domain.InvalidTestFileException
 import org.malachite.estella.task.domain.TaskNotFoundException
 import org.malachite.estella.task.domain.TaskRepository
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 import java.sql.Timestamp
 import java.util.*
 import javax.sql.rowset.serial.SerialBlob
 import javax.sql.rowset.serial.SerialClob
 
 
 @Service
+@Transactional
 class TaskService(
     @Autowired private val taskRepository: TaskRepository,
     @Autowired private val organizationService: OrganizationService,
     @Autowired private val taskStageService: TaskStageService,
     @Autowired private val recruitmentProcessService: RecruitmentProcessService,
-    @Autowired private val interviewService: InterviewService,
     @Autowired private val securityService: SecurityService
 ) : EStellaService<Task>() {
 
     override val throwable: Exception = TaskNotFoundException()
 
-    fun checkDevPassword(organizationUuid: String, password: String) =
-        organizationService.getOrganization(organizationUuid).let {
-            if (!securityService.compareOrganizationWithPassword(it, password))
-                throw UnauthenticatedException()
-            this
-        }
+    fun checkDevPassword(organizationUuid: UUID?, password: String) =
+        organizationUuid
+            ?.let {
+                organizationService.getOrganization(it)
+            }
+            ?.takeIf {
+                securityService.compareOrganizationWithPassword(it, password)
+            }
+            ?.let { this } ?: throw UnauthenticatedException()
```

Comment:
jakbyś robił jeszcze jakąś wrzutkę, to warto by było zrobić tak, żeby zwracało OrganizationNotFoundException jak nie ma organizcji, a nie zawsze unauthenticated bo moze być zonk na froncie potem

Reply:
Nie wiem czy w przypdaku uuid, mówienie po reście, że nie znaleziono organizacji nie równa się `SECURITY BREACH`

Reply:
Popieram to co @Pinioo  napisał

Reply:
nie jestem jakoś super przekonany że komuś wiedza że jakaś organizacja nie istnieje poszerzy pole do niecnych zamiarów ani że ktoś akurat się wstrzeli i trafi w taką co akurat jest - widzę za to oczyma duszy kurwowanie na froncie jak po godzienie walki z tym dlaczego dałes złe hasło orientujesz się że dałes złe UUID XD

Reply:
W sumie u nas na poziomie requesta organizationUUID zawiera się w haśle deva

Reply:
Znaczy wydaje mi się dużo mniej intuicyjne to że w sytuacji jak podałeś złe hasło dostaniesz przy poście do jakiegoś endpointu z tasków, informacje że organizacja o takim uuid nie istnieje. Bo zaczniesz przeglądać dany endpoint i serwis w poszukiwaniu, gdzie może ci się wywalić coś z organizacjami. A tak wiesz że musisz patrzeć na hasło.

Reply:
dobra jestem zjebany XD nie zauważyłem że to jest sprawdzanie hasła tylko myślałem że to ma coś już na obiektach robić xD